api-channel-jira/src/main/java/com/synopsys/integration/alert/api/channel/jira/distribution/JiraErrorMessageUtility.java

```diff
@@ -33,12 +34,18 @@ public JiraErrorMessageUtility(Gson gson) {
         this.gson = gson;
     }
 
-    public AlertException improveRestException(IntegrationRestException restException, String issueCreatorFieldKey, String issueCreatorEmail) {
+    public AlertException improveRestException(IntegrationRestException restException, String issueCreatorFieldKey, String issueCreatorEmail, JiraCustomFieldResolver jiraCustomFieldResolver) {
         String message = restException.getMessage();
         try {
             List<String> responseErrors = extractErrorsFromResponseContent(restException.getHttpResponseContent(), issueCreatorFieldKey, issueCreatorEmail);
             if (!responseErrors.isEmpty()) {
-                message += " | Details: " + StringUtils.join(responseErrors, ", ");
+                String responseErrorString = StringUtils.join(responseErrors, ", ");
+                if (responseErrorString.contains("customfield_")) {
+                    for (String customFieldId : jiraCustomFieldResolver.getCustomFieldIds()) {
+                        responseErrorString = responseErrorString.replace(customFieldId, jiraCustomFieldResolver.resolveCustomFieldIdToName(customFieldId));
+                    }
+                }
```

Comment:
Could this code move into a separate component that has its own `JiraCustomFieldResolver`?